pkg/kubectl/cmd/run.go

```diff
@@ -88,6 +88,14 @@ var (
 		kubectl run pi --schedule="0/5 * * * ?" --image=perl --restart=OnFailure -- perl -Mbignum=bpi -wle 'print bpi(2000)'`))
 )
 
+type RunObject struct {
+	Object  runtime.Object
+	Mapper  meta.RESTMapper
+	Mapping *meta.RESTMapping
+}
+
+var runObjectMap = map[string]*RunObject{}
```

Comment:
Do not use global vars, if you need to pass this between methods either do it directly, or create a struct methods and update the internal state of it, similarly to how it's done in other commands.